pulsar-client-cpp/lib/HTTPLookupService.cc

```diff
@@ -148,132 +148,148 @@ void HTTPLookupService::handleNamespaceTopicsHTTPRequest(NamespaceTopicsPromise
+        LOG_INFO("Response received for url " << completeUrl << " response_code " << response_code << " curl res " << res);
+
+        // Free header list
+        curl_slist_free_all(list);
+
+        switch (res) {
+            case CURLE_OK:
+                long response_code;
+                curl_easy_getinfo(handle, CURLINFO_RESPONSE_CODE, &response_code);
+                LOG_INFO("Response received for url " << completeUrl << " code " << response_code);
+                if (response_code == 200) {
+                    retResult = ResultOk;
+                } else if (response_code == 307) {
+                    char *url = NULL;
+                    curl_easy_getinfo(handle, CURLINFO_REDIRECT_URL, &url);
+                    LOG_INFO("Response from url " << completeUrl << " to new url " << url);
+                    completeUrl = url;
+                    retResult = ResultLookupError;
+                } else {
+                    retResult = ResultLookupError;
+                }
+                break;
+            case CURLE_COULDNT_CONNECT:
+            case CURLE_COULDNT_RESOLVE_PROXY:
+            case CURLE_COULDNT_RESOLVE_HOST:
+            case CURLE_HTTP_RETURNED_ERROR:
+                LOG_ERROR("Response failed for url " << completeUrl << ". Error Code " << res);
+                retResult = ResultConnectError;
+                break;
+            case CURLE_READ_ERROR:
+                LOG_ERROR("Response failed for url " << completeUrl << ". Error Code " << res);
+                retResult = ResultReadError;
+                break;
+            case CURLE_OPERATION_TIMEDOUT:
+                LOG_ERROR("Response failed for url " << completeUrl << ". Error Code " << res);
+                retResult = ResultTimeout;
+                break;
+            default:
+                LOG_ERROR("Response failed for url " << completeUrl << ". Error Code " << res);
                 retResult = ResultLookupError;
-            }
-            break;
-        case CURLE_COULDNT_CONNECT:
-        case CURLE_COULDNT_RESOLVE_PROXY:
-        case CURLE_COULDNT_RESOLVE_HOST:
-        case CURLE_HTTP_RETURNED_ERROR:
-            LOG_ERROR("Response failed for url " << completeUrl << ". Error Code " << res);
-            retResult = ResultConnectError;
-            break;
-        case CURLE_READ_ERROR:
-            LOG_ERROR("Response failed for url " << completeUrl << ". Error Code " << res);
-            retResult = ResultReadError;
-            break;
-        case CURLE_OPERATION_TIMEDOUT:
-            LOG_ERROR("Response failed for url " << completeUrl << ". Error Code " << res);
-            retResult = ResultTimeout;
-            break;
-        default:
-            LOG_ERROR("Response failed for url " << completeUrl << ". Error Code " << res);
-            retResult = ResultLookupError;
+                break;
+        }
+        curl_easy_cleanup(handle);
+        if (response_code != 307) {
             break;
+        }
```

Comment:
What if the response code is other 30x?

See https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#3xx_redirection